### ISSUE TITLE
Add option to trim directory names

### ISSUE THREAD

**Issue by synox** [RhetTbull/osxphotos#317]
I am using
```
/usr/local/bin/python3 -m osxphotos export \
	--update  --directory "{created.date} {place.name.city,} {place.country_code,}" \
		"/Users/synox/Pictures/Photos Library.photoslibrary" \
		"/Users/synox/Pictures/Apple-Photos-Exported"
```
Which leads to directory names like: 
- `2020-12-23 Zürich CH`
- `2020-12-23  `

The last one has no location, and therefore two leading spaces. 

It would be great to trim them by default, or add an option.

**Comment by RhetTbull** [comment]
Good suggestion.  The easiest would be to trim (`str.strip()` in python) both directories and filenames (or maybe all templates) as this is the most likely thing expected/desired by the user.  There could be cases where someone wants the leading or trailing space but I'd expect those to be a minority.

Another option would be to a '{strip}' template that if present in the template, strips all leading/trailing white space.

Finally, I could add a {place?} boolean template (currently working on adding more of these in issue #289).  This would allow you to achieve the desired result using this template:

`--directory "{created.date}{place? ,}{place.name.city,}{place? ,}{place.country_code,}"`

I'll think about what's the cleanest way to implement this.

**Comment by RhetTbull** [comment]
After looking at the template code, it wouldn't be difficult to add a boolean version of every template value.  e.g. you could do this:

--directory "{created.date}{place.name.city? ,}{place.name.city,}{place.country_code? ,}{place.country_code,}"

**Comment by RhetTbull** [comment]
Partial fix in release [v0.39.5](https://github.com/RhetTbull/osxphotos/releases/tag/v0.39.5)

I added a `--strip` option that, if set, will strip all leading and trailing whitespace from rendered strings.  This applies globally to all templates used in the export session.  

I'll work on extending the boolean template options at a later time.
